k8_cortx_cloud/deploy-cortx-cloud.sh

```diff
@@ -829,35 +873,41 @@ function deployCortxControl()
     printf "# Deploy CORTX Control                                  \n"
     printf "########################################################\n"
 
-    cortxcontrol_image=$(parseSolution 'solution.images.cortxcontrol')
-    cortxcontrol_image=$(echo $cortxcontrol_image | cut -f2 -d'>')
-
-    external_services_type=$(parseSolution 'solution.common.external_services.type')
-    external_services_type=$(echo $external_services_type | cut -f2 -d'>')
-
-    cortxcontrol_machineid=$(cat $cfgmap_path/auto-gen-control-$namespace/id)
-
-    num_nodes=1
-    helm install "cortx-control-$namespace" cortx-cloud-helm-pkg/cortx-control \
+    local control_image
+    local control_service_type
+    local control_service_ports_https
+    local control_machineid
+    control_image=$(getSolutionValue 'solution.images.cortxcontrol')
+    control_service_type=$(getSolutionValue 'solution.common.external_services.control.type')
+    control_service_ports_https=$(getSolutionValue 'solution.common.external_services.control.ports.https')
+    control_machineid=$(cat "${cfgmap_path}/auto-gen-control-${namespace}/id")
+
+    local optional_values=()
+    local control_service_nodeports_https
+    control_service_nodeports_https=$(getSolutionValue 'solution.common.external_services.control.nodePorts.https')
+    [[ -n ${control_service_nodeports_https} ]] && optional_values+=(--set cortxcontrol.service.loadbal.nodePorts.https="${control_service_nodeports_https}")
+
+    helm install "cortx-control-${namespace}" cortx-cloud-helm-pkg/cortx-control \
         --set cortxcontrol.name="cortx-control" \
-        --set cortxcontrol.image=$cortxcontrol_image \
+        --set cortxcontrol.image="${control_image}" \
         --set cortxcontrol.service.loadbal.name="cortx-control-loadbal-svc" \
-        --set cortxcontrol.service.loadbal.type="$external_services_type" \
+        --set cortxcontrol.service.loadbal.type="${control_service_type}" \
+        --set cortxcontrol.service.loadbal.ports.api="${control_service_ports_https}" \
```

Comment:
> loadbal.ports.api

Ok, but why not name this `loadbal.ports.https`?

Reply:
Good question.